tailfs/compositefs/compositefs_test.go

```diff
@@ -0,0 +1,507 @@
+				if err == nil || test.err.Error() != err.Error() {
+					t.Errorf("expected error: %v   got: %v", test.err, err)
+				}
+			} else {
+				if err != nil {
+					t.Errorf("unexpected error: %v", err)
+				} else {
+					fi, err := fs.Stat(ctx, test.newName)
+					if err != nil {
+						t.Errorf("unexpected error: %v", err)
+					} else {
+						// Override modTime to avoid having to compare it
+						test.expectedNewInfo.ModdedTime = fi.ModTime()
+						infosEqual(t, test.expectedNewInfo, fi)
+					}
+				}
+			}
+		})
+	}
+}
+
+func createFileSystem(t *testing.T, opts *Opts) (webdav.FileSystem, string, string, *tstest.Clock, func()) {
+	l1, dir1 := startRemote(t)
+	l2, dir2 := startRemote(t)
+
+	// Make some files, use perms 0666 as lowest common denominator that works
+	// on both UNIX and Windows.
+	err := os.WriteFile(filepath.Join(dir1, "file1.txt"), []byte("12345"), 0666)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = os.WriteFile(filepath.Join(dir2, "file2.txt"), []byte("54321"), 0666)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// make some directories
+	err = os.Mkdir(filepath.Join(dir1, "dir1"), 0666)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = os.Mkdir(filepath.Join(dir2, "dir2"), 0666)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if opts == nil {
+		opts = &Opts{}
+	}
+	if opts.Logf == nil {
+		opts.Logf = t.Logf
+	}
+	clock := tstest.NewClock(tstest.ClockOpts{Start: time.Now()})
+	opts.Clock = clock
+
+	fs := New(opts)
+	fs.AddChild(&Child{Name: "remote4", FS: &closeableFS{webdav.Dir(dir2)}})
+	fs.SetChildren(&Child{Name: "remote2", FS: webdav.Dir(dir2)},
+		&Child{Name: "remote3", FS: &closeableFS{webdav.Dir(dir2)}},
+	)
+	fs.AddChild(&Child{Name: "remote1", FS: webdav.Dir(dir1)})
+	fs.RemoveChild("remote3")
+
+	child, ok := fs.GetChild("remote1")
+	if !ok || child == nil {
+		t.Fatal("unable to GetChild(remote1)")
+	}
+	child, ok = fs.GetChild("remote2")
+	if !ok || child == nil {
+		t.Fatal("unable to GetChild(remote2)")
+	}
+	child, ok = fs.GetChild("remote3")
+	if ok || child != nil {
+		t.Fatal("should have been able to GetChild(remote3)")
+	}
+	child, ok = fs.GetChild("remote4")
+	if ok || child != nil {
+		t.Fatal("should have been able to GetChild(remote4)")
+	}
+
+	return fs, dir1, dir2, clock, func() {
+		defer l1.Close()
+		defer os.RemoveAll(dir1)
+		defer l2.Close()
+		defer os.RemoveAll(dir2)
+	}
+}
+
+func stat(t *testing.T, path string) fs.FileInfo {
+	fi, err := os.Stat(path)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return fi
+}
+
+func startRemote(t *testing.T) (net.Listener, string) {
+	dir, err := os.MkdirTemp("", "")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	l, err := net.Listen("tcp", "127.0.0.1:")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	h := &webdav.Handler{
+		FileSystem: webdav.Dir(dir),
+		LockSystem: webdav.NewMemLS(),
+	}
+
+	s := &http.Server{Handler: h}
+	go s.Serve(l)
+
+	return l, dir
+}
+
+func infosEqual(t *testing.T, expected, actual fs.FileInfo) {
+	if expected.Name() != actual.Name() {
```

Comment:
`t.Helper()`
so that errors attribute to the right LoC when a test fails.

Reply:
Good call.

Reply:
Fixed.